server.js

```diff
@@ -1,33 +1,132 @@
-import express from 'express'
-import cors from 'cors'
+import express from "express";
+import cors from "cors";
 
 // If you're using one of our datasets, uncomment the appropriate import below
 // to get started!
-// 
+//
 // import goldenGlobesData from './data/golden-globes.json'
 // import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
+
+import booksData from "./data/books.json";
+
+// console.log(booksData.length); -- 450st!
+
 // import netflixData from './data/netflix-titles.json'
 // import topMusicData from './data/top-music.json'
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
+// Defines the port the app will run on. Defaults to 8080, but can be
 // overridden when starting the server. For example:
 //
 //   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(express.json())
+app.use(cors());
+app.use(express.json());
 
 // Start defining your routes here
-app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+app.get("/", (req, res) => {
+  res.send("Please use an endpoint to fetch requests!");
+});
+
+//main fetch for all books
+app.get("/books", (req, res) => {
+  const { author, title } = req.query;
+
+  // Started with pagination but didn't get the response I wanted.
+  // Wanted the fetch to be on /books and not ?page=n&limit=n
+
+  // const page = parseInt(req.query.page);
+  // const limit = parseInt(req.query.limit);
+
+  // const startIndex = (page - 1) * limit;
+  // const endIndex = page * limit;
+
+  // const results = {};
+
+  // if (endIndex < booksData.length) {
+  //   results.next = {
+  //     page: page + 1,
+  //     limit: limit,
+  //   };
+  // }
+
+  // if (startIndex > 0) {
+  //   results.previous = {
+  //     page: page - 1,
+  //     limit: limit,
+  //   };
+  // }
+
+  // results.results = booksData.slice(startIndex, endIndex);
+  // res.send(results);
```

Comment:
I realize you probably want to continue working on this but if you are not planing on doing it quite soon it can be very easy to forget about it and then this out commented code will take up space/make the code less clean :)